### ISSUE TITLE
Raiden 0.200.0 rc3 gets 'NoneType' object has no attribute 'hex'

### ISSUE THREAD

**Issue by agatsoh** [raiden-network/raiden#6040]
## Problem Definition
While starting raiden all the nodes the get the same error 
```

Traceback (most recent call last):
  File "raiden/ui/cli.py", line 610, in run
    run_services(kwargs)
  File "raiden/ui/runners.py", line 21, in run_services
    app = run_app(**options)
  File "raiden/ui/app.py", line 438, in run_app
    raiden_app.start()
  File "raiden/app.py", line 91, in start
    self.raiden.start()
  File "raiden/raiden_service.py", line 382, in start
    self._start_transport(chain_state)
  File "raiden/raiden_service.py", line 490, in _start_transport
    raiden_service=self, prev_auth_data=None, health_check_list=health_check_list
  File "raiden/network/transport/matrix/transport.py", line 463, in start
    self._initialize_first_sync()
  File "raiden/network/transport/matrix/transport.py", line 763, in _initialize_first_sync
    timeout_ms=self._config.sync_timeout, latency_ms=self._config.sync_latency
  File "raiden/network/transport/matrix/client.py", line 567, in blocking_sync
    self._handle_responses(pending_queue)
  File "raiden/network/transport/matrix/client.py", line 774, in _handle_responses
    self._handle_member_join_callback(room)
  File "raiden/network/transport/matrix/transport.py", line 1024, in _handle_member_join
    [room], "Users from more than one address joined the room"
  File "raiden/network/transport/matrix/transport.py", line 797, in _leave_unexpected_rooms
    partners=[to_checksum_address(partner) for partner in partners],
  File "raiden/network/transport/matrix/transport.py", line 797, in <listcomp>
    partners=[to_checksum_address(partner) for partner in partners],
  File "raiden/utils/formatting.py", line 54, in to_checksum_address
    v = int.from_bytes(keccak(bytes(address.hex(), "ascii")), byteorder="big")
AttributeError: 'NoneType' object has no attribute 'hex'

```
Most likely guess would be that the matrix servers underwent a restart and hence they were unreachable.

[node-Aa1Ab1-goerli.2ndApril2020.log](https://github.com/raiden-network/raiden/files/4420406/node-Aa1Ab1-goerli.2ndApril2020.log)

**Comment by konradkonrad** [comment]
> "Users from more than one address joined the room"

That seems worth investigating -- https://github.com/raiden-network/raiden/blob/5ce09881cb3452cab4e49c131a6c80cc23b63d99/raiden/network/transport/matrix/transport.py#L1024

We should probably add a testcase with a fading homeserver, to see how we're handling these conditions.

**Comment by Dominik1999** [comment]
I have the same issue when starting RC3 with a node that worked yesterday
```
Traceback (most recent call last):
  File "raiden/ui/cli.py", line 610, in run
  File "raiden/ui/runners.py", line 21, in run_services
  File "raiden/ui/app.py", line 438, in run_app
  File "raiden/app.py", line 91, in start
  File "raiden/raiden_service.py", line 382, in start
  File "raiden/raiden_service.py", line 490, in _start_transport
  File "raiden/network/transport/matrix/transport.py", line 463, in start
  File "raiden/network/transport/matrix/transport.py", line 763, in _initialize_first_sync
  File "raiden/network/transport/matrix/client.py", line 567, in blocking_sync
  File "raiden/network/transport/matrix/client.py", line 774, in _handle_responses
  File "raiden/network/transport/matrix/transport.py", line 1024, in _handle_member_join
  File "raiden/network/transport/matrix/transport.py", line 797, in _leave_unexpected_rooms
  File "raiden/network/transport/matrix/transport.py", line 797, in <listcomp>
  File "raiden/utils/formatting.py", line 54, in to_checksum_address
AttributeError: 'NoneType' object has no attribute 'hex'
```

**Comment by konradkonrad** [comment]
It looks as if https://github.com/raiden-network/raiden/blob/5ce09881cb3452cab4e49c131a6c80cc23b63d99/raiden/network/transport/matrix/transport.py#L897 can return `None` for a partner.

**Comment by Dominik1999** [comment]
- Why does that happen - Display User Name in Matrix seems wrong - investigate that 
- Handle `None` Case
- MyPy seems not to do its work

**Comment by fredo** [comment]
I found the reason for it.

# Description
The error happened after anyblocks was introduced into the federation and especially as the first server in the list.
Once there is a new first server in the list, broadcast rooms of this server will be used in the future. This leaves the node in an old public room with multiple addresses. By definition these rooms (excluding broadcast rooms) should be left automatically once multiple addresses are in a room.
Now the tricky part comes. Besides all the old addresses there are also the admin users of the servers in the room. These do not follow the Displayname/Username scheme.
When the room is inspected by the transport if there are multiple addresses in the room, they get extracted by Displayname. Obviously, the admin users do not have a displayname set.

The good thing: Nothing behaved unexpectedly. The case that an admin can be in the room was just not handled. These are the only users which can be in a room at this point and do not have displayname set. 

# Solution 
This should be an easy one. Handle the NoneType as "one user from a different address" since not having a display name set is an odd behavior in general. 

Im gonna create an PR

**Comment by Dominik1999** [comment]
nice and quick! Thanks!

**Comment by fredo** [comment]
fixed by #6041
